src/lib/engine/utils.js

```diff
@@ -39,3 +39,11 @@ export function makeURLsMarkdown(text) {
 
   return transformedText
 }
+
+export const indicesOf = (list, predFn) => list.map((e, i) => predFn(e) ? i : null).filter(e => _.isFinite(e))
+export function getSubMatches(str, regex, matchIndex) {
+  if (_.isEmpty(str)) { return [] }
+  let matches = [], match
```

Comment:
I think you should declare match on a newline, just because we don't use this pattern elsewhere (from what I can tell)